src/PublishCodeCovCoverageTask/utils/webUtils.ts

```
@@ -10,208 +10,252 @@ import * as url from 'node:url';
+                            // Throttle progress updates
+                            const now = Date.now();
+                            if (now - lastProgressTime >= progressThrottleMs || percent !== lastReportedPercent) {
+                                lastProgressTime = now;
+                                lastReportedPercent = percent;
 
-                file.close((err) => {
-                    if (err) {
-                        fs.unlink(dest, (unlinkErr) => {
-                            if (unlinkErr) {
-                                console.warn(`Warning: Failed to clean up temporary file '${dest}' after file close error: ${unlinkErr.message}`);
+                                options.onProgress!({
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)